path/dijkstra.go

```diff
@@ -38,8 +38,8 @@ func DijkstraFrom(u graph.Node, g graph.Graph) Shortest {
 	for Q.Len() != 0 {
 		mid := heap.Pop(&Q).(distanceNode)
 		k := path.indexOf[mid.node.ID()]
-		if mid.dist < path.dist[k] {
-			path.dist[k] = mid.dist
+		if mid.dist > path.dist[k] {
```

Comment:
This optimisation requires that there be an addendum to the comment above explaining its existence - i.e. that this implementation deviates from the paper.